lightkurve/targetpixelfile.py

```diff
@@ -41,6 +44,110 @@ def __init__(self, path, quality_bitmask='default', targetid=None, **kwargs):
         self.quality_mask = self._quality_mask(quality_bitmask)
         self.targetid = targetid
 
+
+    def get_sources(self, catalog=None, magnitude_limit=18, dist_tolerance=3):
+        """
+        Returns a table of stars that are centered on the target
+        of the tpf file. Current catalog supported are KIC, EPIC and Gaia DR2.
+
+        Parameters
+        -----------
+        catalog: 'KIC', 'EPIC' or 'Gaia'
+            Catalog to query. For Kepler target pixel files the default is 'KIC',
+            for K2 target pixel files the default is 'EPIC'.
+        magnitude_limit : int or float
+            Limit the returned magnitudes to only stars brighter than magnitude_limit.
+            Default 18 for both Kepmag and Gmag.
+        dist_tolerance: float
+            Maximum distance (in arcseconds) a source may be separated from the edge
+            of the target pixel file.
+
+        Returns
+        -------
+        result : astropy.table
+            Astropy table with the following columns
+            ID : Catalog ID from catalog.
+            RA: Right ascension [degrees] (KIC & EPIC)
+            DEC: Declination [degrees]    (KIC & EPIC)
+            RA: Right ascension [degrees] (Gaia)
+            DEC: Declination [degrees] (Gaia)
+            pmRA: Proper motion for right ascension [mas/year]
+            pmDEC: Proper motion for declination [mas/year]
+            Kpmag: Magnitude in Kepler band [mag] (KIC & EPIC)
+            Gmag: Magnitude in Gaia band [mag]  (Gaia)
+            column: Pixel column coordinate [pix]
+            row: Pixel row coordinate [pix]
+            predicted_flux: Predicted flux of source [e-/s]
+            epoch: Epoch of data [years]
+
+        """
+        if catalog is None:
+            if self.mission == 'Kepler':
+                catalog = 'KIC'
+                catalog_epoch = 2000.0
+            elif self.mission == 'K2':
+                catalog = 'EPIC'
+                catalog_epoch = 2000.0
+            else:
+                raise ValueError('Please provide a catalog. Catalog must be KIC, EPIC or Gaia.')
+        else:
+            catalog_epoch = 2015.5
+
+        if dist_tolerance < 3:
+            log.warning('Distance tolerance is less than 3 pixels. Sources near the edge will likely be missed.')
```

Comment:
Can we reword this message?
'Edge tolerance is less than 3 pixels. Bright sources near the edge of the Target Pixel File may not be included.'